src/if_statements.py

```diff
@@ -421,162 +357,220 @@ def gather_any_comma_conditions(block_info: BlockInfo) -> Condition:
-            # self-loop node's conditional edge points to itself.
-            or next_node.loop
-            or len(next_node.parents) > 1
+        if end not in curr_node.fallthrough_edge.postdominators:
+            break
+        chained_cond_nodes.append(curr_node)
+        # If and/or detection is disabled; limit the chain length to 1 node
+        if not context.options.andor_detection:
+            break
+        if not isinstance(curr_node.fallthrough_edge, ConditionalNode):
+            break
+        curr_node = curr_node.fallthrough_edge
+        if curr_node.loop or any(
+            p not in chained_cond_nodes for p in curr_node.parents
         ):
-            # We reached the end of an && statement.
-
-            if bottom is end:
-                # If we don't need to emit an 'else', only emit && conditions up
-                # to the first comma-statement condition, to avoid too much of
-                # the output being sucked into if conditions.
-                index = next(
-                    (
-                        i
-                        for i, cond in enumerate(conditions)
-                        if isinstance(cond, CommaConditionExpr)
-                    ),
-                    None,
-                )
+            break
 
-                if index is not None:
-                    context.emitted_nodes |= set(condition_nodes[:index])
-                    if_body = build_flowgraph_between(
-                        context, condition_nodes[index], end
-                    )
-                    return IfElseStatement(
-                        join_conditions(
-                            [cond.negated() for cond in conditions[:index]], "&&"
-                        ),
-                        if_body=if_body,
-                    )
+    # We want to take the largest chain of ConditionalNodes where each node's edges
+    # are only to: other nodes forward in the chain, the `if_node`, or the `else_node`.
+    # We start with the largest chain computed above, and then trim it until it
+    # meets this criteria. The resulting chain will always have at least one node.
+    if_node: Node
+    else_node: Optional[Node]
+    while True:
+        assert chained_cond_nodes
+        if_node = chained_cond_nodes[-1].fallthrough_edge
+        else_node = chained_cond_nodes[-1].conditional_edge
+
+        # If the else body would be empty (`else_node is end`), then check the nodes
+        # that are part of the outermost && expression, if it exists.
+        # Complex && conditions are better represented by nested ifs
+        if else_node is end:
+            outermost_has_comma = False
+            node: Node = start
+            while node in chained_cond_nodes:
+                assert isinstance(node, ConditionalNode)
+                if node.conditional_edge is end:
+                    block_info = node.block.block_info
+                    assert block_info
```

Comment:
```suggestion
                    assert isinstance(block_info, BlockInfo)
```
otherwise block_info will be Any